OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -577,13 +605,19 @@ void PopulateAssetList()
 				}
 			}
 
-			foreach (var file in files.OrderBy(s => s.Key))
+			foreach (var file in files)
 			{
-				if (!allowedExtensions.Any(ext => file.Key.EndsWith(ext, true, CultureInfo.InvariantCulture)))
-					continue;
-
 				foreach (var package in file.Value)
+				{
+					// Don't show unknown files in the engine dir - it is full of code and git and IDE files.
+					// But do show supported types that are inside just in case.
```

Comment:
I don't really like this, as it is hidden and not visible to the outside at all. Besides, when you select `unknown`, aren't you exactly asking for this? A much better fix would be making the list with packages also one where you can tick/untick packages and maybe have the engine dir unticked by default? (Fwiw, several of the files in the engine dir will already be filtered by the "no extension" rule.)

Reply:
> when you select unknown, aren't you exactly asking for this?

Yes, but I was previously asked to hide engine files so I added this 🤷‍♂️